packages/jbrowse-cli/src/commands/add-connection.ts

```diff
@@ -0,0 +1,317 @@
+import { Command, flags } from '@oclif/command'
+import { promises as fsPromises } from 'fs'
+import * as path from 'path'
+import fetch from 'node-fetch'
+
+interface Connection {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any
+}
+
+interface Config {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  assemblies?: { name: string; sequence: { [key: string]: any } }[]
+  configuration?: {}
+  connections?: Connection[]
+  defaultSession?: {}
+  tracks?: unknown[]
+}
+
+export default class AddConnection extends Command {
+  static description = 'Add a connection to a JBrowse 2 configuration'
+
+  static examples = [
+    '$ jbrowse add-connection http://mysite.com/jbrowse/data/',
+    '$ jbrowse add-connection http://mysite.com/jbrowse/custom_data_folder/ --type JBrowse1Connection',
+    '$ jbrowse add-connection http://mysite.com/path/to/hub.txt --assemblyName hg19',
+    '$ jbrowse add-connection http://mysite.com/path/to/custom_hub_name.txt --type UCSCTrackHubConnection --assemblyName hg19',
+    `$ jbrowse add-connection http://mysite.com/path/to/custom --type custom --config '{"uri":{"url":"https://mysite.com/path/to/custom"}}' --assemblyName hg19`,
+    '$ jbrowse add-connection https://mysite.com/path/to/hub.txt --connectionId newId --name newName',
+  ]
+
+  // if not JBrowse1 or UCSC, they must provide the the config object
+
+  static args = [
+    {
+      name: 'dataDirectory',
+      required: true,
+      description: `URL of data directory\nFor hub file, usually called hub.txt\nFor JBrowse 1, location of JB1 data directory similar to http://mysite.com/jbrowse/data/ `,
+    },
+    {
+      name: 'location',
+      required: false,
+      description: 'Location of JBrowse installation. Defaults to .',
```

Comment:
oclif adds a description of the default if you use the "default" param, so technically you don't need the default in the description here.